src/codec/brotli/decoder.rs

```diff
@@ -4,34 +4,49 @@ use std::{
     io::{Error, ErrorKind, Result},
 };
 
-use brotli2::raw::{DeStatus, Decompress};
+use brotli::{enc::StandardAlloc, BrotliDecompressStream, BrotliResult, BrotliState};
 
 pub struct BrotliDecoder {
-    decompress: Decompress,
+    state: BrotliState<StandardAlloc, StandardAlloc, StandardAlloc>,
 }
 
 impl BrotliDecoder {
     pub(crate) fn new() -> Self {
         Self {
-            decompress: Decompress::new(),
+            state: BrotliState::new(
+                StandardAlloc::default(),
+                StandardAlloc::default(),
+                StandardAlloc::default(),
+            ),
         }
     }
 
     fn decode(
         &mut self,
         input: &mut PartialBuffer<&[u8]>,
         output: &mut PartialBuffer<&mut [u8]>,
-    ) -> Result<DeStatus> {
-        let mut in_buf = input.unwritten();
+    ) -> Result<BrotliResult> {
+        let in_buf = input.unwritten();
         let mut out_buf = output.unwritten_mut();
 
-        let original_input_len = in_buf.len();
-        let original_output_len = out_buf.len();
+        let mut input_len = 0;
+        let mut output_len = 0;
 
-        let status = self.decompress.decompress(&mut in_buf, &mut out_buf)?;
-
-        let input_len = original_input_len - in_buf.len();
-        let output_len = original_output_len - out_buf.len();
+        let status = match BrotliDecompressStream(
+            &mut in_buf.len(),
```

Comment:
I assume the modifications to `available_in`/`available_out` just mirror the modifications to `input_offset`/`output_offset`, so we can ignore them?

Reply:
Yeah, after looking a bit more into how `brotli` works this looks like the case.